### PR TITLE
Fix URIs matching error in sdrfToNfConf.R

### PR DESCRIPTION
**Pull request by pmb59**
The PR fixes  the error for the droplet protocol, that arises when `uri_select` is not a matrix or a data.frame:
```
During startup - Warning message:
Setting LC_CTYPE failed, using "C"
Error in FUN(X[[i]], ...) :
  Can't find URIs matching files: SRR13491863_2.fastq.gz
Calls: lapply -> FUN -> lapply -> FUN
Execution halted
```

The PR now reports a single value (TRUE or FALSE) as expected, as `files` in this case 

```
          } else {
            missing_uri_files <- files[which(! any(apply(species.protocol.sdrf[, uri_cols], 2, function(x) basename(x) == files))  )]
```
has a single file.

Tested with the following accessions: E-GEOD-153723, E-GEOD-157775, E-MTAB-4850

### CODE REVIEW

File: bin/sdrfToNfConf.R [hunk 1]
```diff
@@ -1077,14 +1077,14 @@ configs <- lapply(species_list, function(species){
...
-          uri_select <- apply(species.protocol.sdrf[,uri_cols], 2, function(x) basename(x) == files)         
+          uri_select <- apply(species.protocol.sdrf[,uri_cols], 2, function(x) basename(x) == files)
           
           if (is.data.frame(uri_select) || is.matrix(uri_select)) {
             # the following line gives an error: dim(X) must have a positive length
             # if not a matrix or a data.frame
             missing_uri_files <- files[which(! apply(apply(species.protocol.sdrf[,uri_cols], 2, function(x) basename(x) == files), 1, any))]
           } else {
-            missing_uri_files <- files[which(! apply(as.data.frame( apply(species.protocol.sdrf[,uri_cols], 2, function(x) basename(x) == files) ), 1, any))]
+            missing_uri_files <- files[which(! any(apply(species.protocol.sdrf[, uri_cols], 2, function(x) basename(x) == files))  )]
```
Comment (anilthanki) [inline bin/sdrfToNfConf.R:1087]: Can you please explain briefly what are you trying to do here.. So I know what I understood from code is right..

Comment (pmb59) [reply bin/sdrfToNfConf.R:1087]: Hi @anilthanki  Did you have a look to the description of the PR?

Basically,  in this chunk for each library it is checked if there is a fastq URI that can supply the file. The error was in the fix for the case when `uri_select` is not a matrix or a data.frame

**Review by anilthanki — APPROVED** [review]
LGTM

**Comment by pmb59** [comment]
Thanks for reviewing @anilthanki

**Pull request merged**